doc/dev/release_checklist.md

```diff
@@ -0,0 +1,135 @@
+# Releasing a stable version
+
+This checklist walks you through releasing a stable version.
+See below for the checklist for release candidates.
+
+We'll use `v4.6.0` as the intended release version as a running example.
+
```

Comment:
```suggestion

- One week before the planned release, ensure that someone writes the first draft of the release announcement
```

Reply:
Or maybe this should be part of the RC checklist instead?

Reply:
I'm either misunderstanding, or disagree. The announcement itself doesn't need to happen in advance: I usually write it on the spot before I send it.

Reply:
Ah, I see. By "announcement" I meant the short zulip comment, while you meant the blog post.

Reply:
Yes, sorry - how about this version?

```suggestion

- One week before the planned release, ensure that someone has written the first draft of the release blog post
```